### PR TITLE
Fix PostAttesterDuties to be compliant with spec (min 1 index)

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Spec requires at least one index https://github.com/ethereum/beacon-APIs/blob/master/apis/validator/duties/attester.yaml#L33-L39 which was not checked. Check added + test.

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [ ] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by zilm13** [comment]
I see there are 4 apis with `minItems: 1`, so I will try to implement something more generic

**Comment by zilm13** [comment]
So more generic definition was made and following endpoints fixed:
https://github.com/ethereum/beacon-APIs/blob/master/apis/validator/duties/sync.yaml#L15-L23
https://github.com/ethereum/beacon-APIs/blob/master/apis/validator/duties/attester.yaml#L31-L39
https://github.com/ethereum/beacon-APIs/blob/master/apis/validator/liveness.yaml#L22-L30

File: infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableArrayTypeDefinition.java [hunk 1]
```diff
@@ -28,13 +29,25 @@ public class DeserializableArrayTypeDefinition<ItemT, CollectionT extends Iterab
 
   private final DeserializableTypeDefinition<ItemT> itemType;
   private final Function<List<ItemT>, CollectionT> createFromList;
+  private final Optional<Integer> minItems;
```
Comment (rolfyone) [inline infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableArrayTypeDefinition.java:32]: we should be expanding on `DeserializableArrayTypeDefinitionTest`

**Review by rolfyone — COMMENTED** [review]
overall looks good. Need to expand typeDef tests a little, but shouldn't be too much of a problem.

**Comment by zilm13** [comment]
Added tests

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**